### PR TITLE
Generating clients in build

### LINKED ISSUE #156: update-codegen.sh is broken
```
→ ./hack/update-codegen.sh 
bash: ../code-generator/generate-groups.sh: No such file or directory
```

Once fixed, should be added to `gen-k8s` verb in Makefile.

### PR DESCRIPTION
**Pull request by jakub-dzon**
Fixes #156
```release-note
NONE
```

Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -19,7 +19,14 @@ set -o nounset
 set -o pipefail
 
 SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; ls -d -1 $GOPATH/src/github.com/kubernetes/code-generator 2>/dev/null || echo ../code-generator)}
+if [ ! -d "$GOPATH/src/github.com/kubernetes/code-generator" ]
+then
+    wget https://github.com/kubernetes/code-generator/archive/v0.16.4.tar.gz -P /tmp
```
Comment (masayag) [inline hack/update-codegen.sh]: why not to include it in vendor ?
have a look at tools.go, it includes packages that are required for build/test only:

```
// +build tools

package tools

import (
        _ "github.com/aktau/github-release"
        _ "github.com/onsi/ginkgo/ginkgo"
        _ "github.com/operator-framework/operator-sdk/cmd/operator-sdk"
        _ "golang.org/x/tools/cmd/goimports"
)
```

Comment (jakub-dzon) [reply hack/update-codegen.sh]: I was unable to download that tool with any of go tools. It's always:
`build k8s.io/code-generator: cannot load k8s.io/code-generator: no Go source files`

**Review by pkliczewski — APPROVED** [review]

**Comment by jakub-dzon** [comment]
Eventually it worked.
@machacekondra - thanks for your help!

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**